src/pudl_archiver/archivers/ferc/xbrl.py

```diff
@@ -205,63 +213,62 @@ async def archive_taxonomy(
-            for url in taxonomy.urlDocs:
-                url_parsed = urlparse(url)
-
-                # There are some generic XML/XBRL files in the taxonomy that should be skipped
-                if not url_parsed.netloc.endswith("ferc.gov"):
-                    continue
-
-                # Download file
-                response = await retry_async(session.get, args=[url])
-                response_bytes = await retry_async(response.content.read)
-                path = Path(url_parsed.path).relative_to("/")
-                archive_files.add(path)
-
-                with archive.open(str(path), "w") as f:
-                    f.write(response_bytes)
-
-        resources.append(
-            ResourceInfo(
-                local_path=archive_path,
-                partitions={"year": year, "data_format": "XBRL_TAXONOMY"},
-                layout=ZipLayout(file_paths=archive_files),
+    taxonomy_versions = []
+    archive_path = output_dir / f"ferc{form.as_int()}-xbrl-taxonomies.zip"
+    with zipfile.ZipFile(archive_path, "w", compression=ZIP_DEFLATED) as archive:
+        for taxonomy_entry_point in taxonomies_referenced:
+            logger.info(f"Archiving {taxonomy_entry_point}.")
+
+            # Use Arelle to parse taxonomy
+            cntlr = Cntlr.Cntlr()
+            cntlr.startLogging(logFileName="logToPrint")
+            model_manager = ModelManager.initialize(cntlr)
+            taxonomy = await retry_async(
+                asyncio.to_thread,
+                args=[ModelXbrl.load, model_manager, taxonomy_entry_point],
+                retry_on=(FileNotFoundError, FileExistsError),
             )
-        )
-    return resources
+
+            # Loop through all files and save to appropriate location in archive
+            archive_files = set()
+            taxonomy_buffer = io.BytesIO()
+            with zipfile.ZipFile(
+                taxonomy_buffer, "w", compression=ZIP_DEFLATED
+            ) as taxonomy_archive:
+                for url in taxonomy.urlDocs:
+                    url_parsed = urlparse(url)
+
+                    # There are some generic XML/XBRL files in the taxonomy that should be skipped
+                    if not url_parsed.netloc.endswith("ferc.gov"):
+                        continue
+
+                    # Download file
+                    response = await retry_async(session.get, args=[url])
+                    response_bytes = await retry_async(response.content.read)
+                    path = Path(url_parsed.path).relative_to("/")
+                    archive_files.add(path)
+
+                    with taxonomy_archive.open(str(path), "w") as f:
+                        f.write(response_bytes)
+
+            taxonomy_version = TAXONOMY_URL_PATTERN.match(taxonomy_entry_point).group(1)
+            taxonomy_versions.append(taxonomy_version)
+            with archive.open(f"{taxonomy_version}.zip", mode="w") as taxonomy_zip:
```

Comment:
A nit: right now the file names look like `form-6_2023-04-01.zip`. Can we get them to be `form-6-2023-04-01.zip` instead to spare having to use both `-` and `_`?

Reply:
I kept the formatting from the ferc url's because taxonomies are referenced by URL, and I use these in the extractor to figure out which version to use when parsing filings. Wouldn't be hard to do the `_` to `-` change on the extractor side as well though, so I'm open to making the change.